src/Calf/PhaseDistinction.agda

```diff
@@ -62,3 +62,26 @@ postulate
     (h : (a : val A) → (u : ext) → P.subst (λ a → cmp (X a)) (η≡∗ a u) (x0 a) ≡ x1 u ) →
     ●/ind (∗ u) X x0 x1 h ≡ x1 u
   {-# REWRITE ●/ind/β₂ #-}
+
+
+record Extension (X : tp neg) (φ : Ω) (spec : (u : φ) → cmp X) : Set where
+  field
+    out : cmp X
+    law : (u : φ) → out ≡ spec u
+open Extension public
+
+witnessed-by : {X : tp neg} {φ : Ω} {out : cmp X} {spec : (u : φ) → cmp X} → ((u : φ) → out ≡ spec u) → Extension X φ spec
+witnessed-by {out = out} h = record { out = out ; law = h }
+
+exactly : {X : tp neg} {φ : Ω} {spec : cmp X} → Extension X φ λ _ → spec
+exactly {spec = spec} = record { out = spec ; law = λ u → refl }
+
+postulate
+  extension-≡ : ∀ {X φ spec} {x y : Extension X φ spec} → out x ≡ out y → x ≡ y
+  -- Trivial proof if `law` field is irrelevant. However, this is annoying to
+  -- work with, since it requires that further-up proofs be irrelevant.
```

Comment:
Is there a better way to do this? Per the comment, I tried
```agda
.law : (u : φ) → out ≡ spec u
```
(note the `.` before `law`, denoting [irrelevance](https://agda.readthedocs.io/en/v2.6.3/language/irrelevance.html)), but this requires that all applications are irrelevant, cluttering the use site of `law` (in the proof of `backward` in `classic-amortization`).